
From: Arjan van de Ven <arjanv@redhat.com>

It fixes old X servers whcih want to read kernel memory looking for ROM
signatures.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/char/mem.c |    2 --
 1 files changed, 2 deletions(-)

diff -puN drivers/char/mem.c~dev-mem-restriction-patch-allow-reads drivers/char/mem.c
--- 25/drivers/char/mem.c~dev-mem-restriction-patch-allow-reads	2004-09-24 02:09:34.911284392 -0700
+++ 25-akpm/drivers/char/mem.c	2004-09-24 02:09:34.914283936 -0700
@@ -187,8 +187,6 @@ static ssize_t read_mem(struct file * fi
 		}
 	}
 #endif
-	if (!range_is_allowed(p, p+count))
-		return -EPERM;
 	if (copy_to_user(buf, __va(p), count))
 		return -EFAULT;
 	read += count;
_
